internal/appsec/remoteconfig.go

```diff
@@ -48,13 +48,13 @@ func mergeMaps[K comparable, V any](m1 map[K]V, m2 map[K]V) map[K]V {
 }
 
 // combineRCRulesUpdates updates the state of the given rulesManager with the combination of all the provided rules updates
-func combineRCRulesUpdates(r *rulesManager, updates map[string]remoteconfig.ProductUpdate) (map[string]rc.ApplyStatus, error) {
-	statuses := map[string]rc.ApplyStatus{}
+func (r *rulesManager) combineRCRulesUpdates(updates map[string]remoteconfig.ProductUpdate) (statuses map[string]rc.ApplyStatus, err error) {
+	statuses = make(map[string]rc.ApplyStatus, len(updates))
```

Comment:
Just FYI the final allocated space will be even greater than that since each update can hold several configs, and each status is related to one config.

Reply:
I guess here we are good to avoid re-allocs assuming there's exactly one config per update (which I reckon there is at least); and we'll reallocate only once we exceed that, which I think is still fewer allocs than before... Although a benchmark would have helped determine that ^^